lib/ElementHandle.js

```diff
@@ -341,6 +341,24 @@ class ElementHandle extends JSHandle {
     }
     return result;
   }
+
+  /**
+   * @returns {!Promise<boolean>}
+   */
+  isDisplayed() {
+    return this.executionContext().evaluate(
+        node => new Promise(resolve => {
+          const callback = entries => {
+            resolve(entries[0].isIntersecting);
+            observer.unobserve(node);
+          };
+          const observer = new IntersectionObserver(callback, { root: null, rootMargin: '0px', threshold: 1.0 });
```

Comment:
so when will the `callback` be called? Is it always called initially?

Reply:
Yes, it will be called initially and then unobserve from the node